wandb/__init__.py

```diff
@@ -1293,7 +1293,7 @@ def set_global_config(run):
     telemetry_updated = run.config._telemetry_update()
 
     if sagemaker_config:
-        run.config._update(sagemaker_config)
+        run.config._update(sagemaker_config, allow_val_change=allow_val_change)
```

Comment:
I'm not sure if this `if` condition is required. Maybe it can be collapsed along with the next `if` condition?